spec/controllers/api/openid_connect/authorizations_controller_spec.rb

```diff
@@ -0,0 +1,228 @@
+      end
+
+      context "when client id is missing" do
+        it "should return an bad request error" do
+          post :new, redirect_uri: "http://localhost:3000/", response_type: "id_token",
+               scope: "openid", nonce: SecureRandom.hex(16), state: SecureRandom.hex(16)
+          expect(response.body).to match("bad_request")
+        end
+      end
+
+      context "when redirect uri is missing" do
+        context "when only one redirect URL is pre-registered" do
+          it "should return a form page" do
+            # Note this intentionally behavior diverts from OIDC spec http://openid.net/specs/openid-connect-core-1_0.html#AuthRequest
+            # When client has only one redirect uri registered, only that redirect uri can be used. Hence,
+            # we should implicitly assume the client wants to use that registered URI.
+            # See https://github.com/nov/rack-oauth2/blob/master/lib/rack/oauth2/server/authorize.rb#L63
+            post :new, client_id: client.client_id, response_type: "id_token",
+                 scope: "openid", nonce: SecureRandom.hex(16), state: SecureRandom.hex(16)
+            expect(response.body).to match("Diaspora Test Client")
+          end
+        end
+      end
+
+      context "when multiple redirect URLs are pre-registered" do
+        it "should return an invalid request error" do
+          post :new, client_id: client_with_multiple_redirects.client_id, response_type: "id_token",
+               scope: "openid", nonce: SecureRandom.hex(16), state: SecureRandom.hex(16)
+          expect(response.body).to match("bad_request")
+        end
+      end
+
+      context "when redirect URI does not match pre-registered URIs" do
+        it "should return an invalid request error" do
+          post :new, client_id: client.client_id, redirect_uri: "http://localhost:2000/",
+               response_type: "id_token", scope: "openid", nonce: SecureRandom.hex(16)
+          expect(response.body).to match("bad_request")
+        end
+      end
+
+      context "when an unsupported scope is passed in" do
+        it "should return an invalid scope error" do
+          post :new, client_id: client.client_id, redirect_uri: "http://localhost:3000/", response_type: "id_token",
+               scope: "random", nonce: SecureRandom.hex(16), state: SecureRandom.hex(16)
+          expect(response.body).to match("error=invalid_scope")
+        end
+      end
+
+      context "when nonce is missing" do
+        it "should return an invalid request error" do
+          post :new, client_id: client.client_id, redirect_uri: "http://localhost:3000/",
+               response_type: "id_token", scope: "openid", state: SecureRandom.hex(16)
+          expect(response.location).to match("error=invalid_request")
+        end
+      end
+    end
+    context "when already authorized" do
+      let!(:auth) { Api::OpenidConnect::Authorization.find_or_create_by(o_auth_application: client, user: alice,
```

Comment:
Avoid using {...} for multi-line blocks.<br>Block body expression is on the same line as the block start.